flexmeasures/data/schemas/forecasting/pipeline.py

```diff
@@ -208,6 +212,9 @@ def resolve_config(self, data: dict, **kwargs) -> dict:  # noqa: C901
             if predict_period_in_hours < 1:
                 raise click.BadParameter("--predict-period must be at least 1 hour")
 
+        if data["start_date"] is None:
+            start_date = predict_start - timedelta(hours=train_period_in_hours)
```

Comment:
If `data["start_date"] is None`, but `data["train_period"] is not None`, this will crash I think, on the `train_period_in_hours` variable not being defined.

Reply:
I withdraw my comment. It's a falsehood.